src/databricks/labs/lsql/dashboards.py

```diff
@@ -37,111 +51,612 @@
-        parser.add_argument("-h", "--height", type=int)
-        return parser
+    @classmethod
+    def from_path(cls, path: Path) -> "TileMetadata":
+        tile_metadata = cls(path=path)
+        header = tile_metadata.handler.parse_header()
+        header["path"] = path
+        return cls.from_dict(header)
 
-    def replace_from_arguments(self, arguments: list[str]) -> "WidgetMetadata":
-        replica = copy.deepcopy(self)
-        parser = self._get_arguments_parser()
+    def __repr__(self):
+        return f"TileMetadata<{self._path}>"
+
+
+@dataclass
+class DashboardMetadata:
+    display_name: str
+    tiles: dict[str, TileMetadata] = dataclasses.field(default_factory=dict)
+
+    @classmethod
+    def from_dict(cls, raw: dict) -> "DashboardMetadata":
+        display_name = raw["display_name"]  # Fail early if missing
+        tiles, tiles_raw = {}, raw.get("tiles", {})
+        for tile_id, tile_raw in tiles_raw.items():
+            if not isinstance(tile_raw, dict):
+                logger.warning(f"Parsing invalid tile metadata in dashboard.yml: tiles.{tile_id}.{tile_raw}")
+                continue
+            tile = TileMetadata(_id=tile_id)
+            for tile_key, tile_value in tile_raw.items():
+                if tile_key == "id":
+                    logger.warning(f"Parsing unsupported field in dashboard.yml: tiles.{tile_id}.id")
+                    continue
+                try:
+                    tile |= TileMetadata.from_dict({tile_key: tile_value})
+                except TypeError:
+                    logger.warning(f"Parsing unsupported field in dashboard.yml: tiles.{tile_id}.{tile_key}")
+                    continue
+            tiles[tile.id] = tile
+        return cls(display_name=display_name, tiles=tiles)
+
+    def as_dict(self) -> dict:
+        raw: dict = {"display_name": self.display_name}
+        if self.tiles:
+            raw["tiles"] = {tile.id: tile.as_dict() for tile in self.tiles.values()}
+        return raw
+
+    @classmethod
+    def from_path(cls, path: Path) -> "DashboardMetadata":
+        """Export dashboard metadata from a YAML file."""
+        fallback_metadata = cls(display_name=path.parent.name)
+        if not path.exists():
+            return fallback_metadata
         try:
-            args = parser.parse_args(arguments)
-        except (argparse.ArgumentError, SystemExit) as e:
-            logger.warning(f"Parsing {arguments}: {e}")
-            return replica
+            raw = yaml.safe_load(path.read_text())
+        except yaml.YAMLError as e:
+            logger.warning(f"Parsing {path}: {e}")
+            return fallback_metadata
+        try:
+            return cls.from_dict(raw)
+        except KeyError as e:
+            logger.warning(f"Parsing {path}: {e}")
+            return fallback_metadata
+
+
+class Tile:
+    """A dashboard tile."""
+
+    def __init__(self, tile_metadata: TileMetadata) -> None:
+        self._tile_metadata = tile_metadata
```

Comment:
```suggestion
        self._metadata = tile_metadata
```
DRY: we're already in tile.